rag/structured_rag/entity_graph.py

```
@@ -0,0 +1,931 @@
+                events=char_info.get("events", [])
+            )
+            self.relationships.append(relationship)
+
+    def _apply_bidirectional_rules_from_schema(self):
+        """Apply bidirectional relationship rules based on schema configuration."""
+        # Get bidirectional rules from schema
+        bidir_rules = []
+
+        for schema_name, schema_class in self.schema_classes.items():
+            rel_fields = schema_class.get_relationship_fields()
+            for field_name, config in rel_fields.items():
```

Ruff (0.15.9)

[warning] Loop control variable `field_name` not used within loop body

Rename unused `field_name` to `_field_name`

(B007)